src/main/java/mate/academy/dao/BookDaoImpl.java

```diff
@@ -0,0 +1,133 @@
+    private static final String NO_RECORDS_EXCEPTION = "There are no records in DB";
+    private static final int LESS_THAN_ONE = 1;
+    private static final int AT_LEAST_ONE = 0;
+    private static final int LESS_THAN_ZERO = 0;
+    private static final int ZERO = 0;
+
+    @Override
+    public Book create(Book book) {
+        if (book.getTitle() == null || book.getPrice() == null) {
+            throw new RuntimeException(NULL_FIELDS_EXCEPTION);
+        }
+        String sqlUpdate = "INSERT INTO books (title, price) VALUES (?, ?)";
+        try (Connection connection = ConnectionUtil.getConnection();
+             PreparedStatement statement = connection.prepareStatement(sqlUpdate, Statement.RETURN_GENERATED_KEYS)) {
+            statement.setString(1, book.getTitle());
+            statement.setBigDecimal(2, book.getPrice());
+            int affectedRows = statement.executeUpdate();
+            if (affectedRows < LESS_THAN_ONE) {
+                throw new RuntimeException(AT_LEAST_ONE_ROW_EXCEPTION);
+            }
+            ResultSet generatedKeys = statement.getGeneratedKeys();
+            if (generatedKeys.next()) {
+                Long id = generatedKeys.getObject(1, Long.class);
+                book.setId(id);
+            }
+            return book;
+        } catch (SQLException e) {
+            throw new DataProcessingException(ADD_BOOK_EXCEPTION + book, e);
+        }
+    }
+
+    @Override
+    public Optional<Book> findById(Long id) {
+        if (id < LESS_THAN_ZERO) {
+            throw new RuntimeException(NEGATIVE_ID_EXCEPTION);
+        }
+        String sqlQuery = "SELECT * FROM books WHERE id = ?";
+        try (Connection connection = ConnectionUtil.getConnection();
+             PreparedStatement statement = connection.prepareStatement(sqlQuery)) {
+            statement.setLong(1, id);
+            ResultSet resultSet = statement.executeQuery();
+            if (resultSet.next()) {
+                String model = resultSet.getString("title");
+                BigDecimal price = resultSet.getObject("price", BigDecimal.class);
+                return Optional.of(new Book(id, model, price));
+            }
+            throw new RuntimeException(NO_BOOK_EXCEPTION + id);
+        } catch (SQLException e) {
+            throw new DataProcessingException(GET_BOOK_EXCEPTION, e);
+        }
+    }
+
+        @Override
+    public List<Book> findAll() {
+        List<Book> books = new ArrayList<>();
+        String sqlQuery = "SELECT * FROM books";
+        try (Connection connection = ConnectionUtil.getConnection();
+             PreparedStatement statement = connection.prepareStatement(sqlQuery)) {
+            ResultSet resultSet = statement.executeQuery();
+            while (resultSet.next()) {
+                Long id = resultSet.getObject("id", Long.class);
+                String model = resultSet.getString("title");
+                BigDecimal price = resultSet.getObject("price", BigDecimal.class);
```

Comment:
You use this logic to parse book fields multiple times. Put it in a separate private method, something like
private Book parseBook(ResultSet resultSet)
      return new Book(fields...)